### PR TITLE
fix: Thread safety regarding metrics

### PR DESCRIPTION
**Pull request by aleksabl**
Fixed a couple of issues regarding metrics when
`MetricsBucket.registerCount` is called concurrently

### CODE REVIEW

File: src/main/java/no/finn/unleash/metric/ToggleCount.java [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
+        this.yes = new AtomicLong(0);
+        this.no = new AtomicLong(0);
+        this.variants = new ConcurrentHashMap<>();
     }
 
     public void register(boolean active) {
-        if(active) {
-            yes++;
+        if (active) {
+            yes.incrementAndGet();
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/metric/ToggleCount.java]: wow, and I really remember now that `++` is not an atomic operation.

**Comment by ivarconr** [comment]
Thanks, it's nice to get the metric counting thread-safe!

Before merging this I would like to do some performance evaluation. 

Do you have any thoughts on performance @aleksabl ?

**Comment by aleksabl** [comment]
If you create micro benchmark I'm sure you'll find that the code will be a bit slower than the version that only works in a single thread environment. But in the grand scheme of things, I think the performance implications are negible.

**Comment by ivarconr** [comment]
:+1:  I'm sure it probably does not matter, but nice to now what we are talking 1ms vs. 0.003ms

**Comment by aleksabl** [comment]
I'm not sure how much slower it is, but we're definetly not taking about 1ms (or even 0.1ms) here

**Comment by ivarconr** [comment]
I have tested the change (on my machine) it it seems do add about 20ns. I think this is fine and worth it as the accuracy of metrics will be increased. 

If we later want it to be faster we could simply just add results to a list, and perform the counting as part of metrics pusher.

**Pull request merged**